src/common/string_utils.cpp

```diff
@@ -99,5 +99,31 @@ std::string StringUtils::removeEscapedCharacters(const std::string& input) {
     return resultStr;
 }
 
+// Jenkins hash function: https://en.wikipedia.org/wiki/Jenkins_hash_function
```

Comment:
Be more specific here that we transform each character to its lower case and apply Jenkins hash function on converted lower case characters.